patients/repo.go

```diff
@@ -1295,70 +1375,208 @@ func PatientsToTideResult(patientsList []*Patient, period string, exclusions *[]
+				SummaryFieldComparator:      "$gte",
+				ComparatorOperandExpression: 0.7,
+			},
+			{
+				SummaryField:                "timeCGMUsePercent",
+				SummaryFieldComparator:      "$gte",
+				ComparatorOperandExpression: 0.7,
+			},
+			// These "$not" expressions are because the fields may not exist in the patient summary
+			{
+				SummaryField:                "timeInAnyLowPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .04},
+			},
+			{
+				SummaryField:                "timeInVeryLowPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .01},
+			},
+			{
+				SummaryField:                "timeInHighPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .25},
+			},
+			{
+				SummaryField:                "timeInVeryHighPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .05},
+			},
+			{
+				SummaryField:                "timeInExtremeHighPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .01},
+			},
+		},
+	},
+	{
+		CategoryName:          "timeInExtremeHighPercent",
+		SummaryField:          "timeInExtremeHighPercent",
+		SummaryFieldSortOrder: -1,
+		SummaryFieldFilters: []summaryFieldFilter{
+			{
+				SummaryField:                "timeInExtremeHighPercent",
+				SummaryFieldComparator:      "$gt",
+				ComparatorOperandExpression: 0.01,
+			}},
+	},
+	{
+		CategoryName:          "timeInVeryHighPercent",
+		SummaryField:          "timeInVeryHighPercent",
+		SummaryFieldSortOrder: -1,
+		SummaryFieldFilters: []summaryFieldFilter{
+			{
+				SummaryField:                "timeInVeryHighPercent",
+				SummaryFieldComparator:      "$gt",
+				ComparatorOperandExpression: 0.05,
+			}},
+	},
+	{
+		CategoryName:          "timeInHighPercent",
+		SummaryField:          "timeInHighPercent",
+		SummaryFieldSortOrder: -1,
+		SummaryFieldFilters: []summaryFieldFilter{
+			{
+				SummaryField:                "timeInHighPercent",
+				SummaryFieldComparator:      "$gt",
+				ComparatorOperandExpression: 0.25,
+			}},
+	},
+}
+
+// defaultOrderedCategoryNames is the ORDERED listing of categories by name to
+// use if TideReportParams.Categories is empty. It is ordered because some
+// categories have a higher priority in terms of whether patients within that
+// category should be shown or not due to the restriction on returned results
+// by TideReportPatientLimit
+var defaultOrderedCategoryNames = []string{
+	"timeInVeryLowPercent",
+	"timeInAnyLowPercent",
+	"dropInTimeInTargetPercent",
+	"timeInTargetPercent",
+	"timeCGMUsePercent",
+	"meetingTargets",
+}
+
+// getCategoriesByNames returns a slice of categories with a CategoryName in names. The order of the returned slice matches the order of the name in names.
+func getCategoriesByNames(names []string) []tideCategory {
```

Comment:
```suggestion
// categoriesByNames returns a slice of categories whose CategoryName is found in names. The order of the returned slice matches the order of the name in names.
func categoriesByNames(names []string) []tideCategory {
```

https://go.dev/doc/effective_go#Getters